modules/tinymce/src/plugins/paste/main/ts/core/SmartPaste.ts

```diff
@@ -31,8 +31,9 @@ const isAbsoluteUrl = function (url: string) {
   return /^https?:\/\/[\w\?\-\/+=.&%@~#]+$/i.test(url);
 };
 
-const isImageUrl = function (url: string) {
-  return isAbsoluteUrl(url) && /.(gif|jpe?g|png)$/.test(url);
+const isImageUrl = function (editor: Editor, url: string) {
+  const validImageFileTypes = new RegExp('.(' + Settings.allowedImageFileTypes(editor).join('|') + ')$');
```

Comment:
Are we concerned something might be added to the settings that would conflict with the regex syntax? I'm just wondering if we need to escape the settings here.

Reply:
Will add a regex escape function to katamari